crates/sdl-encoder/src/schema_def.rs

```diff
@@ -0,0 +1,112 @@
+use crate::Field;
+use std::fmt::{self, Display};
+
+/// A GraphQL service’s collective type system capabilities are referred to as that service’s “schema”.
+///
+/// *SchemaDefinition*:
+///     Description<sub>opt</sub> **schema** Directives<sub>\[Const\] opt</sub> **{** RootOperationTypeDefinition<sub>list</sub> **}**
+///
+/// Detailed documentation can be found in [GraphQL spec](https://spec.graphql.org/draft/#sec-Schema).
+///
+/// ### Example
+/// ```rust
+/// use sdl_encoder::{FieldValue, Field, SchemaDef};
+/// use indoc::indoc;
+///
+/// let ty_1 = FieldValue::Type {
+///     ty: "TryingToFindCatQuery".to_string(),
+/// };
+///
+/// let field = Field::new("query".to_string(), ty_1);
+///
+/// let mut schema_def = SchemaDef::new(field);
+/// schema_def.description(Some("Root Schema".to_string()));
+///
+/// assert_eq!(
+///     schema_def.to_string(),
+///     indoc! { r#"
+///         """Root Schema"""
+///         schema {
+///           query: TryingToFindCatQuery
+///         }
+///     "#}
+/// );
+/// ```
+
+#[derive(Debug, Clone)]
+pub struct SchemaDef {
+    // Description may be a String.
+    description: Option<String>,
+    // The vector of fields in a schema to represent root operation type
+    // definition.
+    fields: Vec<Field>,
```

Comment:
Since these aren't real fields (they can't have arguments) but references to root types, it might make more sense to include these as `query_type`, `mutation_type`, and `subscription_type` (as is done in `graphql-js` and other libraries).

Reply:
Ah, ok!

Am I meant to encode the schema definition then based on the following values? 
```json
            "queryType": {
                "name": "Query"
            },
            "mutationType": null,
            "subscriptionType": null,
```

Reply:
Yep, the introspection query only asks for the `name`, which should be enough (there is no need for spreading a `TypeRef` fragment here because root types are always object types, never wrapped in `List` or `NonNull`).

Reply:
oh thank god 🙏 at least one of these is straightforward